Content/Items/Beach/Acessories.SaltCogs.cs

```diff
@@ -0,0 +1,89 @@
+﻿using StarlightRiver.Content.Items.BaseTypes;
+using StarlightRiver.Content.Items.Food;
+using Terraria.ID;
+
+namespace StarlightRiver.Content.Items.Beach
+{
+	class SaltCogs : SmartAccessory
+	{
+		public override string Texture => AssetDirectory.Assets + "Items/Beach/" + Name;
+
+		public SaltCogs() : base("Salt Cogs", "Your sentries fire twice as fast for two seconds after placing them\n100% increased sentry placement speed") { }
+
+		public override void Load()
+		{
+			StarlightProjectile.PostAIEvent += SpeedUp;
+			StarlightProjectile.PostDrawEvent += DrawCog;
+			StarlightItem.UseTimeMultiplierEvent += PlaceSpeed;
+			StarlightItem.UseAnimationMultiplierEvent += PlaceAnimation;
+		}
+
+		public override void SafeSetDefaults()
+		{
+			Item.rare = ItemRarityID.Blue;
+			Item.value = Item.sellPrice(silver: 50);
+		}
+
+		private void SpeedUp(Projectile projectile)
+		{
+			Player owner = Main.player[projectile.owner];
+
+			if (owner != null && Equipped(owner))
+			{
+				if (projectile.timeLeft == (Projectile.SentryLifeTime - 1))
+					projectile.extraUpdates += 1;
+
+				if (projectile.timeLeft == (Projectile.SentryLifeTime - 240))
```

Comment:
isn't 240 4 seconds while the description says 2 seconds? I'd say move this 240 into a const and have the number in the description be programmatically assigned

Reply:
oh wait does extraUpdates make SentryLifeTime count more times too? that would explain this

Reply:
yep, 120 would be 1 second as extraUpdates causes it to tick down twice as fast.